deployment/data-streams/jobs/stream.go

```diff
@@ -1 +1,93 @@
 package jobs
+
+import (
+	"bytes"
+	"text/template"
+
+	"github.com/pelletier/go-toml/v2"
+)
+
+type Datasource struct {
+	BridgeName string
+	ReqData    string
+}
+
+type ReportField struct {
+	ResultPath string
+}
+
+type ObservationSource struct {
+	Datasources   []Datasource
+	AllowedFaults int
+	Benchmark     ReportField
+	Bid           ReportField
+	Ask           ReportField
+}
+
+type LLOSpec struct {
+	Base
+
+	StreamID          string `toml:"streamID"`
+	ObservationSource string `toml:"observationSource,multiline,omitempty"`
+}
+
+func (s *LLOSpec) SetObservationSource(obs ObservationSource) error {
+	rendered, err := s.buildObservationSource(obs)
+	if err != nil {
+		return err
+	}
+	s.ObservationSource = rendered
+	return nil
+}
+
+func (s *LLOSpec) buildObservationSource(obs ObservationSource) (string, error) {
+	var buf bytes.Buffer
+	if err := observationTmpl.Execute(&buf, obs); err != nil {
+		return "", err
+	}
+	return buf.String(), nil
+}
+
+func (s *LLOSpec) MarshalTOML() ([]byte, error) {
+	return toml.Marshal(s)
+}
+
+var funcMap = template.FuncMap{
+	"inc": func(i int) int {
+		return i + 1
+	},
+}
+
+var pipelineTemplate = `{{range $i, $a := .Datasources}}
```

Comment:
we should put this into a file

Reply:
Changed the logic a little bit and updated this as well!